From: =?utf-8?q?I=C3=B1igo_Mart=C3=ADnez?= <inigomartinez@gmail.com>
Date: Sat, 14 Apr 2018 23:06:40 +0200
Subject: build: Fix `USER_DIR_MODE` value in config.h

meson defines `USER_DIR_MODE` with a raw octal value to be used as
the default permissions when creating the user's configuration
directory.

However, meson does not support raw octal values[0], so the define
misses the initial `0` value. Due to this, the directory is created
with wrong permissions.

This has been changed to use the octal value as a string in meson,
so the definition has the proper value.

Fixes #49

[0] https://github.com/mesonbuild/meson/issues/2047


(cherry picked from commit 37a6b940cb83d97b808da77f397e34100beb263f)
---
 meson.build | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 90ee21c..0f84d52 100644
--- a/meson.build
+++ b/meson.build
@@ -41,7 +41,9 @@ foreach define: set_defines
   config_h.set_quoted(define[0], define[1])
 endforeach
 
-config_h.set('USER_DIR_MODE', 0700,
+# meson does not support octal values, so it must be handled as a
+# string. See: https://github.com/mesonbuild/meson/issues/2047
+config_h.set('USER_DIR_MODE', '0700',
              description: 'Permissions for creating the user\'s config, cache and data directories')
 
 # compiler flags
